packages/blade/src/components/Text/Decisions/Decisions.md

```diff
@@ -0,0 +1,87 @@
+The text component is categorised into multiple parts
+
+- BaseText - Internal Component, not published to consumers
+- Title - Public Component
+- Heading - Public Component
+- Text - Public Component
+- Code - Public Component
+
+Here's how the hierarchy will looks like
+![Component Breakdown](./component-breakdown.png)
+
+The APIs for all the above component will look something like below
+
+- `BaseText`
+
+  This component is just meant to be used for internal purposes for Blade
+  Here's how the API will look like
+
+  ```jsx
+  <BaseText
+    color="surface.text | feedback.text | button.text"
+    fontFamily="fontfamily.name"
+    fontSize="fontSize.name"
+    fontWeight="fontWeight.name"
+    lineWeight="lineWeight"
+    letterSpacing="letterSpacing"
+    formatting="bold/italic/strike"
+    as="h1-h6/p/span"
+  />
+  ```
+
+- `Title`
```

Comment:
>Can we not have this as a variant of Heading?

Actually no because they have different meaning. Titles are mostly the texts that are free form on the landing pages, the big texts you saw vs Heading which always are like section breakpoints and will be followed by paragraph or subheadings.


>Also can this be confused with HTML's <title>? 

hmm, interesting. any other suggestions you might suggest? you can look the hierarchy diagram and understand the semantic behind each component and propose any suggestion for Title?

Reply:
> Actually no because they have different meaning. Titles are mostly the texts that are free form on the landing pages, the big texts you saw vs Heading which always are like section breakpoints and will be followed by paragraph or subheadings.

It can still be variant of Heading right? For example, it still is a 'Heading' of the page. Having paragraph isn't a requirement right? Even in terms of semantics, it will be `<h1 />`.

I think Vercel has `<Text title>` component for their page headings, Chakra uses `<Heading />` itself.


If we still want separate components, I can think of these names - `<H1 />`, `<ScreenTitle />`, `<PageTitle />`, `<MainHeading />`.

Reply:
Re: variant of heading - it can be a variant of heading but as I mentioned we would want the internal system to take care of that instead of consumers making these choices that what HTML heading tag to use. so once we have semantic named components like Title, Heading, Text, Code which dictates the intent and what to render internally is the decision left up to the underlying system

Re: Title - I think we shall start thinking from interfaces in general and not map it to native HTML as we have audiences such as designers and devs from different backgrounds like React native. so Sticking with title for now.